redis-lettuce/src/main/java/io/micronaut/configuration/lettuce/cache/RedisCache.java

```diff
@@ -230,12 +278,47 @@ public void put(Object key, Object value) {
         putValue(serializedKey, value);
     }
 
+    public void put(Map<?, ?> values) {
+        if (CollectionUtils.isNotEmpty(values)) {
+            Map<byte[], byte[]> toSave = new HashMap<>(values.size());
+            Map<byte[], Long> toExpire = expireAfterWritePolicy != null ? new HashMap<>(values.size()) : null;
+            List<byte[]> toDelete = new ArrayList<>();
```

Comment:
pass `values.size()` as well

Reply:
My assumption was that most of the time puts will be non-nulls, so allocating a fully-sized list here seemed wasteful.  I can change it if you still think it would be better to allocate up front.